### PR TITLE
Change ViewSets to actual view sets

### PR DESCRIPTION
**Pull request by MattWindsor91**
Minor change, eliminates a spurious use of multisets, seems to work.

Also removes `View`: nothing uses it.

### CODE REVIEW

File: Main.fs [hunk 1]
```diff
@@ -517,7 +517,7 @@ let runStarling request =
         | Request.Z3 rq       -> phase (symproof >> proof approx >> z3 reals rq) Response.Z3
         | Request.HSF         -> phase (filterIndefinite >> hsf) Response.HSF
         | Request.MuZ3 rq     -> phase (filterIndefinite >> muz3 reals rq) Response.MuZ3
-        | Request.SymZ3 rq    -> phase (symproof >> (tuplize (proof approx >> z3 reals rq))) Response.SymZ3 
+        | Request.SymZ3 rq    -> phase (symproof >> (tuplize (proof approx >> z3 reals rq))) Response.SymZ3
```
Comment (MattWindsor91) [inline Main.fs:520]: Oops, this wasn't supposed to be part of the commit...

**Comment by bensimner** [comment]
Would it be worth merging the different view types (`OView`/`DView`) into a single `View` type? I'm assuming that was the original intention of `View`.

**Comment by MattWindsor91** [comment]
@BenSimner Not sure how you would—some are `List`s, some are `Multiset`s.  What would be nice is changing the names to be more descriptive.  For example, `DView` is really a `ViewSignature`, and `OView` could be `OrderedView`.

**Comment by bensimner** [comment]
What is the point of an ordered view here then?

**Comment by MattWindsor91** [comment]
@BenSimner I think ordered views were required to make the matching logic in reification happen, but I can't remember.  When @mjp41 returns we can ask him?  (I think it was his change, and I forget the rationale).

**Comment by MattWindsor91** [comment]
Incidentally, it might also be worth either renaming `GView` to `GuardedView`, _or_ deleting it in favour of eg. `View<GuardedFunc<'var>>`.  These types might be getting a bit out of hand at this point though.  I think it'd be worthwhile to make it clearer that `GView` puts the guard on the _func_, not the view itself.

**Comment by MattWindsor91** [comment]
We need a bonfire of the type aliases soon.

**Comment by bensimner** [comment]
Soon

**Comment by MattWindsor91** [comment]
@septract Is this ready to merge?

**Pull request merged**